app/models/request/sample_compound_aliquot_transfer.rb

```diff
@@ -10,25 +10,53 @@ module Request::SampleCompoundAliquotTransfer
   # at the source has a different tag_depth defined
   def compound_samples_needed?
     return false if asset.aliquots.count == 1
-    (asset.aliquots.pluck(:tag_depth).uniq.count == asset.aliquots.count)
+
+    _tag_clash?
   end
 
   # Creates a sample in a single aliquot at destination as a compound sample
   # where the component samples are all samples at source
-  def transfer_aliquots_into_compound_sample_aliquot
-    compound_sample = _create_compound_sample(_default_compound_study, asset.samples)
+  def transfer_aliquots_into_compound_sample_aliquots
+    _aliquots_by_tags_combination.each do |_tags_combo, aliquot_list|
+      transfer_into_compound_sample_aliquot(aliquot_list)
+    end
+  end
+
+  private
+
+  def transfer_into_compound_sample_aliquot(source_aliquots)
+    samples = source_aliquots.map(&:sample)
+
+    if source_aliquots.pluck(:tag_depth).uniq.count != source_aliquots.size
+      raise "Cannot create compound sample from following samples due to duplicate 'tag depth': #{samples.map(&:name)}"
+    end
+
+    compound_sample = _create_compound_sample(_default_compound_study, samples)
+
+    add_aliquot(compound_sample, source_aliquots.first.tag_id, source_aliquots.first.tag2_id)
+  end
+
+  def add_aliquot(sample, tag_id, tag2_id)
     target_asset
       .aliquots
-      .create(sample: compound_sample)
+      .create(sample: sample)
       .tap do |aliquot|
+        aliquot.tag_id = tag_id
+        aliquot.tag2_id = tag2_id
         aliquot.library_type = _default_library_type
         aliquot.study_id = _default_compound_study.id
         aliquot.project_id = _default_compound_project_id
         aliquot.save
       end
   end
 
-  private
+  def _tag_clash?
```

Comment:
Maybe change this name? The method isn't really checking if there is a tag_clash I don't think? But the size of the aliquots for one tag combo?

Reply:
Yeah it kind of is, because if there are 2 aliquots with the same tag combination, then the tags clash. 
But Eduardo found a similar bit of code confusing, so I will have a go at renaming the methods 👍